session_manager.py

```diff
@@ -34,4 +39,5 @@ def delete_old_sessions(self):
         for sess in self.sessions:
             if not sess.is_expired():
                 continue
+            self.r.set(sess.get_key(), sess.to_json())
```

Comment:
Remove the session first and then write it to redis, otherwise you can have a race condition.